### ISSUE TITLE
Is `BaseChecker.__gt__` required

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pylint#6045]
### Bug description

As noted by @DanielNoord [here](https://github.com/PyCQA/pylint/pull/5938#discussion_r837867526), [`BaseCheck.__gt__`](https://github.com/PyCQA/pylint/blob/742e60dc07077cdd3338dffc3bb809cd4c27085f/pylint/checkers/base_checker.py#L62-L64) is not currently covered. If this required then we should add a unit test, otherwise we can remove this method.

### Configuration

```ini
N/A
```

### Command used

```shell
N/A
```

### Pylint output

```shell
N/A
```

### Expected behavior

N/A

### Pylint version

```shell
N/A
```

**Comment by Pierre-Sassoulas** [comment]
I think this was used originally to be able to assert that  a list of checker is equal to another one in tests. If it's not covered it means we do not do that anymore.

**Comment by Pierre-Sassoulas** [comment]
It's used in Sphinx and maybe downstream libraries see #6047 .

**Comment by DanielNoord** [comment]
Shall we add a no coverage param then?

**Comment by jpy-git** [comment]
It's pretty easy to add a unit test for this so will make a quick PR
